src/obstacle_params_parser.cpp

```diff
@@ -0,0 +1,172 @@
+        Eigen::Map<Eigen::Vector3d>(translation_vect.data());
+
+    std::vector<double> rotation_vect;
+    if (!pnh_->hasParam(obstacle_name + "/rotation")) {
+      std::cerr << "No obstacle rotation declared." << std::endl;
+      return;
+    }
+    pnh_->getParam(obstacle_name + "/rotation", rotation_vect);
+    const auto rotation = Eigen::Map<Eigen::Quaterniond>(rotation_vect.data());
+
+    pinocchio::GeometryObject::CollisionGeometryPtr geometry;
+
+    if (type == "sphere") {
+      double radius;
+      if (pnh_->hasParam(obstacle_name + "/radius")) {
+        pnh_->getParam(obstacle_name + "/radius", radius);
+        geometry = pinocchio::GeometryObject::CollisionGeometryPtr(
+            new hpp::fcl::Sphere(radius));
+      } else {
+        std::cerr << "No dimension or wrong dimensions in the obstacle "
+                     "config. Try to use "
+                     "the ones for the shapes desired, such as 'radius' for "
+                     "'sphere','x', 'y', 'z' for 'box' or 'halfLength' and "
+                     "'radius' for 'capsule'."
+                  << std::endl;
+        return;
+      }
+    } else if (type == "box") {
+      double x;
+      double y;
+      double z;
+      if (pnh_->hasParam(obstacle_name + "/x") &&
+          pnh_->hasParam(obstacle_name + "/y") &&
+          pnh_->hasParam(obstacle_name + "/z") ) {
+        pnh_->getParam(obstacle_name + "/x", x);
+        pnh_->getParam(obstacle_name + "/y", y);
+        pnh_->getParam(obstacle_name + "/z", z);
+        geometry = pinocchio::GeometryObject::CollisionGeometryPtr(
+            new hpp::fcl::Box(x, y, z));
+      } else {
+        std::cerr << "No dimension or wrong dimensions in the obstacle "
+                     "config. Try to use "
+                     "the ones for the shapes desired, such as 'radius' for "
+                     "'sphere','x', 'y', 'z' for 'box' or 'halfLength' and "
+                     "'radius' for 'capsule'."
+                  << std::endl;
+        return;
+      }
+
+    } else if (type == "capsule") {
+      double radius;
+      double halfLength;
+      if (pnh_->hasParam(obstacle_name + "/radius") &&
+          pnh_->hasParam(obstacle_name + "/halfLength")) {
+        pnh_->getParam(obstacle_name + "/radius", radius);
+        pnh_->getParam(obstacle_name + "/halfLength", halfLength);
+        geometry = pinocchio::GeometryObject::CollisionGeometryPtr(
+            new hpp::fcl::Capsule(radius, halfLength));
+      }
+    } else {
+      std::cerr << "No type or wrong type in the obstacle config. Try to use "
+                   "the one implemented, such as 'sphere', 'box' or 'capsule'."
+                << std::endl;
+      return;
+    }
+    pinocchio::SE3 obstacle_pose(rotation, translation);
+    pinocchio::GeometryObject obstacle(obstacle_name, 0, 0, geometry,
+                                       obstacle_pose);
+    collision_model_->addGeometryObject(obstacle);
+    obs_idx += 1;
```

Comment:
```suggestion
    obs_idx++;
```
This is the proper C/C++ syntax for incrementation